docs/brillig/00_intro.md

```diff
@@ -0,0 +1,47 @@
+# Intro to Brillig
+BRILLIG - Bytecode Runtime Interface for Liberated Language Implementation... G
+
+The G stands for whatever you like, like [Gabagool](https://en.wikipedia.org/wiki/Gabagool).
+
+Noir can compile to bytecode for Brillig, a general VM architecture for use with Noir.
+It is used for functions that are unconstrained, non-deterministic, or otherwise need a VM. 
+
+Why we need Brillig
+---
+
+Noir is a zero-knowledge language that specializes in zero-knowledge proofs that flexibly can compile code to constraints. Despite its specialization, Noir still requires a general VM for the following reasons:
+
+1) Unconstrained execution
+
+In Noir, not all execution situations are fully constrained.
+For example, multiple encrypted notes are a common way to represent single private values.
+When notes need to be presented for a proof, the notes selected need to be verified, but the exact choice is not fully constrained. 
+The note selection algorithm does not need a proof of execution, just the note outputs suffice to constrain the system.
+We refer to these discretionary choice functions as unconstrained functions. 
+```
+// Say we need to reveal and nullify certain notes to pass this check
+// How can we formalize this partially constrained choice?
+if get_balance() > min_amount {
+    // ... perform some action ...
+}
+```
+
+Brillig provides a way to evaluate such unconstrained functions in a safe, general VM.
+
+2) Attributing incorrectness
+
+Brillig's bytecode is used when compiling Noir into a program commitment must differentiate between runtime errors and simply invalid proofs. This proves useful in environments that rely on hedged trust and are distributed, such as blockchains.
+
+In such systems, sequenced proofs of shared public state have unique trust assumptions. Proofs of correctness are not constructed by the user/transactor, creating complications. Two scenarios must be distinguished when executing functions on shared public state:
```

Comment:
We should introduce the public state w/ more of a transition. It reads a bit out of nowhere. 

At the end of the section, it also still isn't clear as to why we need Brillig to achieve this task. Why not use WASM or any other VM? I'm not sure until I read "SNARK proving efficiency" but this is a single phrase and most will miss it.